Image Capture and Edit Improvements

paduerst opened the pull request:
The changes in this PR complete the following item from the feature wishlist:

- Have the Edit Item page allow you to add/replace a picture

To do that, I significantly refactored the `ImageCapture` component and the Edit page. I've also added some more helpful info to the README for developers, and I've added some route protections for the Edit routes.

I've tested these changes on my local machine (MacOS) on Firefox.

paduerst committed:
Added README instructions for making a user an admin

paduerst committed:
Added dev option for manually entering photo URL

paduerst committed:
Fixed broken buttons on Edit page

paduerst committed:
Protecting edit routes now

paduerst committed:
Refactored ImageCapture and edit page

rthorpeii commented on src/components/imageCaptureButtons.tsx:
```diff
@@ -0,0 +1,152 @@
+
+        const file = files[0];
+        file.arrayBuffer().then(arrBuf => {
+            const imgBlob = new Blob([arrBuf], { type: "iamge/jpeg" });
```
```suggestion
            const imgBlob = new Blob([arrBuf], { type: "image/jpeg" });
```

rthorpeii commented on src/components/imageCaptureButtons.tsx:
```diff
@@ -0,0 +1,152 @@
+                    className="px-3 py-1 border border-amber-500 rounded-lg"
+                    type="button"
+                    onClick={() => {
+                        setTakePhoto(true);
+                        setUploadPhoto(false);
+                        setManuallyEnterPhotoUrl(false);
```
Since these are all exclusive, it would be cleaner to do this with a single enum.

paduerst replied:
I agree! Done.

rthorpeii commented on src/components/imageCaptureButtons.tsx:
```diff
@@ -0,0 +1,152 @@
+            {takePhoto && <><Webcam
+                audio={false}
+                ref={webcamRef}
+                height={1920}
+                screenshotFormat="image/jpeg"
+                width={1080}
```
Do we need to specify `height` and `width` explicitly? They're also defined in videoConstraints

paduerst replied:
I suspect we don't need to specify them explicitly. This is copy-pasted from the existing code, and it's been working so far. I'm not interested in doing the testing to confirm we don't need them though, as I'd need to set up an image bucket to upload to for testing.

rthorpeii committed:
typo fix

paduerst committed:
Updated ImageCaptureButtons state to enum

Per Robert's suggestion in my PR, I've refactored the 3 separate boolean states into 1 enum state.
